engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -121,17 +121,44 @@ public NadelExecutionStrategy(List<Service> services,
         List<CompletableFuture<OneServiceExecution>> result = new ArrayList<>();
         for (MergedField mergedField : fieldSubSelection.getMergedSelectionSet().getSubFieldsList()) {
             ExecutionStepInfo fieldExecutionStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(executionCtx, mergedField, rootExecutionStepInfo);
-            Service service = getServiceForFieldDefinition(fieldExecutionStepInfo.getFieldDefinition());
-
-            CreateServiceContextParams parameters = CreateServiceContextParams.newParameters()
-                    .from(executionCtx)
-                    .service(service)
-                    .executionStepInfo(fieldExecutionStepInfo)
-                    .build();
-
-            CompletableFuture<Object> serviceContextCF = serviceExecutionHooks.createServiceContext(parameters);
-            CompletableFuture<OneServiceExecution> serviceCF = serviceContextCF.thenApply(serviceContext -> new OneServiceExecution(service, serviceContext, fieldExecutionStepInfo));
-            result.add(serviceCF);
+            boolean isNamespaced = !fieldExecutionStepInfo.getFieldDefinition().getDirectives("namespaced").isEmpty();
+            if (isNamespaced) {
+                GraphQLObjectType namespacedObjectType = (GraphQLObjectType) fieldExecutionStepInfo.getUnwrappedNonNullType();
```

Comment:
whenever we make an assumption like this - we should have an `Assert`

yes it will blow up with ClassCastException if is not a GraphQLObjectType but an assert is clearer if this should happen